storage_service/locations/models/pipeline.py

```diff
@@ -126,3 +127,53 @@ def create_default_locations(self, shared_path=None):
                     p['purpose'], location, self)
                 LocationPipeline.objects.get_or_create(
                     pipeline=self, location=location)
+
+    # HTTP API CALLS
+
+    def _request_api(self, method, path, **kwargs):
+        url = 'http://{}/api/{}'.format(
+            self.remote_name,
+            path.lstrip('/'),
+        )
+        fields = {
+            'username': self.api_username,
+            'api_key': self.api_key
+        }
+        if 'fields' in kwargs:
+            fields.update(kwargs['fields'])
+
+        LOGGER.debug('URL: %s; fields: %s', url, fields)
+        try:
+            response = requests.request(method, url, data=fields, allow_redirects=True)
+        except requests.exceptions.RequestException as e:
+            LOGGER.exception('Unable to connect to pipeline %s', self)
+            raise e
```

Comment:
A bare `raise` in an exception will re-raise it, which looks like what you're trying to do.  Details: http://www.ianbicking.org/blog/2007/09/re-raising-exceptions.html